File: external/HDF-EOS/HDF4.2r1/mfhdf/ncgen/test0.f
      program fgennc
      include 'netcdf.inc'
      integer  iret
* netCDF id
      integer  ncid
* dimension ids
      integer  idim, jdim, kdim, ldim
* variable ids
      integer  bearsid, orderid, shotid, aloanid, crossid, iid, jid, lid
* variable shapes
      integer dims(3)
* corners and edge lengths
      integer corner(3), edges(3)
* data variables
      character*18 bears
      integer*2 order(3,2)
      integer shot(3,2)
      real aloan(3,2)
      double precision cross(3,2)
      integer i(2)
      real j(3)
      integer*2 l(3)
* attribute vectors
      integer*2  shortval(1)
      integer  nclongval(1)
      real  floatval(3)
      double precision  doubleval(2)
* enter define mode
      ncid = nccre ('ftest0.nc', NCCLOB, iret)
* define dimensions
      idim = ncddef(ncid, 'i', 2, iret)
      jdim = ncddef(ncid, 'j', 3, iret)
      kdim = ncddef(ncid, 'k', NCUNLIM, iret)
      ldim = ncddef(ncid, 'l', 3, iret)
* define variables
      dims(3) = idim
      dims(2) = jdim
      dims(1) = ldim
      bearsid = ncvdef (ncid, 'bears', NCCHAR, 3, dims, iret)
      dims(2) = idim
      dims(1) = jdim
      orderid = ncvdef (ncid, 'order', NCSHORT, 2, dims, iret)
      dims(2) = idim
      dims(1) = jdim
      shotid = ncvdef (ncid, 'shot', NCLONG, 2, dims, iret)
      dims(2) = idim
      dims(1) = jdim
      aloanid = ncvdef (ncid, 'aloan', NCFLOAT, 2, dims, iret)
      dims(2) = idim
      dims(1) = jdim
      crossid = ncvdef (ncid, 'cross', NCDOUBLE, 2, dims, iret)
      dims(1) = idim
      iid = ncvdef (ncid, 'i', NCLONG, 1, dims, iret)
      dims(1) = jdim
      jid = ncvdef (ncid, 'j', NCFLOAT, 1, dims, iret)
      dims(1) = ldim
      lid = ncvdef (ncid, 'l', NCSHORT, 1, dims, iret)
* assign attributes
      call ncaptc(ncid, bearsid, 'act', NCCHAR, 16, 'text string'//char(
     110)//char(9)//'123', iret)
      shortval(1) = -40
      call ncapt(ncid, bearsid, 'acs', NCSHORT, 1, shortval, iret)
      nclongval(1) = 17000
      call ncapt(ncid, bearsid, 'acl', NCLONG, 1, nclongval, iret)
      floatval(1) = -2
      floatval(2) = 1
      floatval(3) = 0
      call ncapt(ncid, bearsid, 'acf', NCFLOAT, 3, floatval, iret)
      doubleval(1) = -1
      doubleval(2) = 0.75
      call ncapt(ncid, bearsid, 'acd', NCDOUBLE, 2, doubleval, iret)
      call ncaptc(ncid, NCGLOBAL, 'history', NCCHAR, 136, 'This is an ex
     1ample of a multi-line global'//char(10)//'attribute.  It could be 
     2used for representing the'//char(10)//'processing history of the d
     3ata, for example.', iret)
* leave define mode
      call ncendf(ncid, iret)
* store bears
      corner(1) = 1
      corner(2) = 1
      corner(3) = 1
      edges(1) = 3
      edges(2) = 3
      edges(3) = 2
      bears = 'indistinguishable' // char(0)
      call ncvptc(ncid, bearsid, corner, edges, bears, 18, iret)
* store order
      corner(1) = 1
      corner(2) = 1
      edges(1) = 3
      edges(2) = 2
      data order /1, 2, 3, 4, 5, 6/
      call ncvpt(ncid, orderid, corner, edges, order, iret)
* store shot
      corner(1) = 1
      corner(2) = 1
      edges(1) = 3
      edges(2) = 2
      data shot /2, 3, 4, 5, 6, 7/
      call ncvpt(ncid, shotid, corner, edges, shot, iret)
* store aloan
      corner(1) = 1
      corner(2) = 1
      edges(1) = 3
      edges(2) = 2
      data aloan /3, 4, 5, 6, 7, 1e+12/
      call ncvpt(ncid, aloanid, corner, edges, aloan, iret)
* store cross
      corner(1) = 1
      corner(2) = 1
      edges(1) = 3
      edges(2) = 2
      data cross /4., 5., 0.000244140625, 7., 8., 10000000000./
      call ncvpt(ncid, crossid, corner, edges, cross, iret)
* store i
      corner(1) = 1
      edges(1) = 2
      data i /10, 20/
      call ncvpt(ncid, iid, corner, edges, i, iret)
* store j
      corner(1) = 1
      edges(1) = 3
      data j /2, 4, 6/
      call ncvpt(ncid, jid, corner, edges, j, iret)
* store l
      corner(1) = 1
      edges(1) = 3
      data l /10, 9, 8/
      call ncvpt(ncid, lid, corner, edges, l, iret)
      call ncclos (ncid, iret)
      end

File: external/HDF-EOS/HDF4.2r1/mfhdf/ncgen/test0.c
#include "netcdf.h"

int
main() {			/* create ctest0.nc */

   int  ncid;			/* netCDF id */

   /* dimension ids */
   int  i_dim, j_dim, k_dim, l_dim;

   /* variable ids */
   int  bears_id, order_id, shot_id, aloan_id, cross_id, i_id, j_id, l_id;

   /* variable shapes */
   int dims[3];

   /* containers for scalar attributes */
   short  short_val;
   nclong  nclong_val;

   /* attribute vectors */
   float  bears_acf[3];
   double  bears_acd[2];

   /* enter define mode */
   ncid = nccreate("ctest0.nc", NC_CLOBBER);

   /* define dimensions */
   i_dim = ncdimdef(ncid, "i", 2L);
   j_dim = ncdimdef(ncid, "j", 3L);
   k_dim = ncdimdef(ncid, "k", NC_UNLIMITED);
   l_dim = ncdimdef(ncid, "l", 3L);

   /* define variables */

   dims[0] = i_dim;
   dims[1] = j_dim;
   dims[2] = l_dim;
   bears_id = ncvardef (ncid, "bears", NC_CHAR, 3, dims);

   dims[0] = i_dim;
   dims[1] = j_dim;
   order_id = ncvardef (ncid, "order", NC_SHORT, 2, dims);

   dims[0] = i_dim;
   dims[1] = j_dim;
   shot_id = ncvardef (ncid, "shot", NC_LONG, 2, dims);

   dims[0] = i_dim;
   dims[1] = j_dim;
   aloan_id = ncvardef (ncid, "aloan", NC_FLOAT, 2, dims);

   dims[0] = i_dim;
   dims[1] = j_dim;
   cross_id = ncvardef (ncid, "cross", NC_DOUBLE, 2, dims);

   dims[0] = i_dim;
   i_id = ncvardef (ncid, "i", NC_LONG, 1, dims);

   dims[0] = j_dim;
   j_id = ncvardef (ncid, "j", NC_FLOAT, 1, dims);

   dims[0] = l_dim;
   l_id = ncvardef (ncid, "l", NC_SHORT, 1, dims);

   /* assign attributes */
   ncattput (ncid, bears_id, "act", NC_CHAR, 16, (void *)"text string\n\t123");
   short_val = -40;
   ncattput (ncid, bears_id, "acs", NC_SHORT, 1,(void *) &short_val);
   nclong_val = 17000;
   ncattput (ncid, bears_id, "acl", NC_LONG, 1,(void *) &nclong_val);
   bears_acf[0] = -2;
   bears_acf[1] = 1;
   bears_acf[2] = 0;
   ncattput (ncid, bears_id, "acf", NC_FLOAT, 3, (void *) bears_acf);
   bears_acd[0] = -1;
   bears_acd[1] = 0.75;
   ncattput (ncid, bears_id, "acd", NC_DOUBLE, 2, (void *) bears_acd);
   ncattput (ncid, NC_GLOBAL, "history", NC_CHAR, 136, (void *)"This is an example of a multi-line global\nattribute.  It could be used for representing the\nprocessing history of the data, for example.");

   /* leave define mode */
   ncendef (ncid);

   {			/* store bears */
    static long bears_start[] = {0, 0, 0};
    static long bears_edges[] = {2, 3, 3};
    static char bears[] = {"indistinguishable"};
    ncvarput(ncid, bears_id, bears_start, bears_edges, (void *)bears);
   }

   {			/* store order */
    static long order_start[] = {0, 0};
    static long order_edges[] = {2, 3};
    static short order[] = {1, 2, 3, 4, 5, 6};
    ncvarput(ncid, order_id, order_start, order_edges, (void *)order);
   }

   {			/* store shot */
    static long shot_start[] = {0, 0};
    static long shot_edges[] = {2, 3};
    static nclong shot[] = {2, 3, 4, 5, 6, 7};
    ncvarput(ncid, shot_id, shot_start, shot_edges, (void *)shot);
   }

   {			/* store aloan */
    static long aloan_start[] = {0, 0};
    static long aloan_edges[] = {2, 3};
    static float aloan[] = {3, 4, 5, 6, 7, 1e+12};
    ncvarput(ncid, aloan_id, aloan_start, aloan_edges, (void *)aloan);
   }

   {			/* store cross */
    static long cross_start[] = {0, 0};
    static long cross_edges[] = {2, 3};
    static double cross[] = {4., 5., 0.000244140625, 7., 8., 10000000000.};
    ncvarput(ncid, cross_id, cross_start, cross_edges, (void *)cross);
   }

   {			/* store i */
    static long i_start[] = {0};
    static long i_edges[] = {2};
    static nclong i[] = {10, 20};
    ncvarput(ncid, i_id, i_start, i_edges, (void *)i);
   }

   {			/* store j */
    static long j_start[] = {0};
    static long j_edges[] = {3};
    static float j[] = {2, 4, 6};
    ncvarput(ncid, j_id, j_start, j_edges, (void *)j);
   }

   {			/* store l */
    static long l_start[] = {0};
    static long l_edges[] = {3};
    static short l[] = {10, 9, 8};
    ncvarput(ncid, l_id, l_start, l_edges, (void *)l);
   }
   ncclose (ncid);
   return 0;
}
